### PR TITLE
Rubocop config in this project root is interfering with rubocop config in apps built during CI

### PR DESCRIPTION
**Pull request by eoinkelly**
### The problem

When we added `.rubocop.yml` to this repo, we accidentally broke rubocop running on the generated apps in CI. The apps are generated into `tmp/builds` so when we ran rubocop in them, rubocop includes the `.rubocop.yml` from this repo's root.

The result is that rubocop runs on 0 files as part of app generation and CI.

### Give rubocop an explicit config file

There does not seem to be a way to tell rubocop not to search parent directories for config so instead we explicitly pass the config file to rubocop when we invoke it. We do this both when we run rubocop to fixup code as part of app generation and when we run it in CI.

### Rubocop behaves differently under different versions of Ruby

The rubocop config for this repo targets Ruby 2.7 (the oldest that Rails 7 supports) via `TargetRubyVersion`. This seems like an appropriate choice for this repo so that we don't generate code that requires Ruby 3.0+.

The Rubocop config in the generated apps does not have a `TargetRubyVersion` specified so it targets whatever version of Ruby the app is running under. This also seems like an appropriate choice.

However this can create conflicts in how some cops are handled. In particular https://docs.rubocop.org/rubocop/cops_lint.html#lintnondeterministicrequireorder changes depending on the Ruby version. We hit this cop on

```ruby
# spec/rails_helper.rb
Dir[Rails.root.join("spec/support/**/*.rb")].sort.each { |f| require f }
# rubocop running on ruby 2.7 requires the sort to be there.
# rubocop running on ruby 3.0+ requires the sort to not be there
```

This can be resolved automatically via `-A` (safe+unsafe) autocorrect but not `-a` (safe only) autocorrect.

I have found `-A` autocorrect safe during the years I have used it on my own code so I think it should be ok to run  it on our generated code here. 

If we prefer not to use `-A` then we could instead disable that rule in the rubocop config for **this repo**.

### CODE REVIEW

File: bin/autoformat [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-rubocop -a -c rubocop.yml
+rubocop -A -c .rubocop.yml
```
Comment (eoinkelly) [inline bin/autoformat:5]: This has been wrong for a long time and nobody noticed. Should we stop generating this file?

File: spec/helpers/retina_image_helper_spec.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
   before { allow(helper).to receive(:image_path) { |file| "/images/#{file}" } }
 
-  it { assert_match(%r{\A<img src=".*" srcset=".*" alt=".*" />\z}, tag) }
+  it { assert_match(%r{\A<img src=".*" srcset=".*" alt=".*"\s*\/?>}, tag) }
```
Comment (eoinkelly) [inline spec/helpers/retina_image_helper_spec.rb:8]: I'm not sure why but the Rails `tag` helper seems to sometimes generate ` />`, ` >`, `>` at different times. This isn't the first time we have had to adjust this spec but hopefully we have all the cases now.

File: bin/ci-run [hunk 3]
```diff
@@ -10,7 +10,7 @@ cd $APP_DIR
 echo "* ******************************************************"
 echo "* Running rubocop"
 echo "* ******************************************************"
-bundle exec rubocop -D --parallel
```
Comment (eoinkelly) [inline bin/ci-run:13]: `-D and --parallel` are on by default

File: bin/autoformat (hunk 1 above)
Comment (eoinkelly) [inline bin/autoformat:5]: This has been broken for years and nobody noticed. Maybe we don't need this file at all?

File: spec/helpers/retina_image_helper_spec.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
   before { allow(helper).to receive(:image_path) { |file| "/images/#{file}" } }
 
-  it { assert_match(%r{\A<img src=".*" srcset=".*" alt=".*" />\z}, tag) }
+  it { assert_match(%r{\A<img src=".*" srcset=".*" alt=".*"\s*/?>}, tag) }
```
Comment (eoinkelly) [inline spec/helpers/retina_image_helper_spec.rb:8]: Not sure what happened here but it seems that sometimes the Rails tag helper ends with ` />` and sometimes it's `>`. I stopped caring and just made the regex support both

File: bin/autoformat (hunk 1 above)
Comment (G-Rath) [reply bin/autoformat:5]: I'm fine with axing it

**Review by G-Rath — APPROVED** [review]
It does make me a little sad that rubocop doesn't support a `root` key like ESLint does, for exactly this situation...

**Comment by robotdana** [comment]
i have a wild suggestion to avoid tho 'it looks at the root rubocop.yml when we don't want it to' problem

add a script for running rubicon that renames the root rubocop file, runs runs rubocop, then renames the file back when it's done

**Comment by eoinkelly** [comment]
>add a script for running rubocop that renames the root rubocop file, runs runs rubocop, then renames the file back when it's done
that seems like more work and maybe a bit more bittle than just being explicit about the config file?

**Comment by eoinkelly** [comment]
@robotdana I'm going to merge this. Happy to chat further in person or in a new issue if you want to explore the meta script idea further

**Pull request merged**